DesignPatternSession2/src/com/metacube/shoppingCart/Dao/InMemoryCartDao.java

```diff
@@ -0,0 +1,86 @@
+package com.metacube.shoppingCart.Dao;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.metacube.shoppingCart.Entity.BaseEntity;
+import com.metacube.shoppingCart.Entity.CartProduct;
+
+/**
+ * Responsible for all the in memory operations of Cart
+ * 
+ * @author Udit Saxena
+ */
+public class InMemoryCartDao implements CartDao {
+	private static InMemoryCartDao cartDao;
```

Comment:
no need to store cart items in dao, every time we can hit the file to get latest data.